src/auth/.gcb/builds/service_account_test/main.tf

```diff
@@ -0,0 +1,83 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+variable "project" {}
+variable "test_runner" {}
+
+data "google_project" "project" {
+}
+
+# The service account we use as our principal for testing service account credentials.
+data "google_service_account" "test-sa-creds-principal" {
+  account_id = "test-sa-creds"
+}
+
+# Generate a key for the test service account credentials.
+resource "google_service_account_key" "test-sa-creds-principal-key" {
+  service_account_id = data.google_service_account.test-sa-creds-principal.name
+}
+
+# This secret stores the ADC json for the principal testing service account credentials.
+resource "google_secret_manager_secret" "test-sa-creds-json-secret" {
+  secret_id   = "test-sa-creds-json"
+  replication {
+    auto {}
+  }
+}
```

Comment:
I assume you are going to rotate this secret somehow?

Reply:
Tracked in #926